airflow/api_fastapi/core_api/routes/public/dags.py

```diff
@@ -199,6 +202,49 @@ def get_dag_details(dag_id: str, session: SessionDep, request: Request) -> DAGDe
     return dag_model
 
 
+@dags_router.get(
+    "/{dag_id}/versions",
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_404_NOT_FOUND,
+        ]
+    ),
+)
+def get_dag_versions(dag_id: str, session: SessionDep, request: Request) -> DagVersionResponse:
+    dag_version = request.app.state.DagVersion()
+    if dag_version is None:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND, detail=f"DagVersion with dag_id {dag_id} was not found"
+        )
+    version_name = dag_version.version
```

Comment:
We are not handling the `~` wildcard dag_id parameters to retrieve for 'all' dags.